scripts/run.sh

```diff
@@ -233,10 +237,50 @@ if [[ ${E2E} == true ]]; then
     --network-runner-grpc-endpoint="0.0.0.0:12342" \
     --avalanchego-path=${AVALANCHEGO_PATH} \
     --avalanchego-plugin-dir=${AVALANCHEGO_PLUGIN_DIR} \
+    --vm-genesis-path=$BASEDIR/genesis.json \
     --output-path=$BASEDIR/avalanchego-${VERSION}/output.yaml \
-    --mode=${MODE}
+    --mode=test
 
   EXIT_CODE=$?
+
+elif [[ ${SIMULATOR} == true ]]; then
+  #################################
+  echo "building e2e.test for simulator"
+  # to install the ginkgo binary (required for test build and run)
+  go install -v github.com/onsi/ginkgo/v2/ginkgo@v2.1.4
+  ACK_GINKGO_RC=true ginkgo build ./tests/e2e
+
+  #################################
+  # By default, it runs all e2e test cases!
+  # Use "--ginkgo.skip" to skip tests.
+  # Use "--ginkgo.focus" to select tests.
+  echo "running e2e.test for simulator"
+  ./tests/e2e/e2e.test \
+    --ginkgo.v \
+    --network-runner-log-level debug \
+    --network-runner-grpc-endpoint="0.0.0.0:12342" \
+    --avalanchego-path=${AVALANCHEGO_PATH} \
+    --avalanchego-plugin-dir=${AVALANCHEGO_PLUGIN_DIR} \
+    --vm-genesis-path=$BASEDIR/genesis.json \
+    --output-path=$BASEDIR/avalanchego-${VERSION}/output.yaml \
+    --mode=run
+  cat $BASEDIR/avalanchego-${VERSION}/output.yaml
+
+  #################################
+  echo "building simulator"
+  pushd ./cmd/simulator
+  go install -v .
+  popd
+  simulator \
+  --network-runner-output=$BASEDIR/avalanchego-${VERSION}/output.yaml \
+  --config=./cmd/simulator/.simulator/config.yml \
+  --keys=./cmd/simulator/.simulator/keys \
+  --timeout=30s \
+  --concurrency=10 \
+  --base-fee=25 \
+  --priority-fee=1
+  EXIT_CODE=$?
```

Comment:
what does this line do?

Reply:
Was meant to be the one for ginkgo, for later kill commands. Let me fix

Reply:
Actually used for at the end of bash script to fail the script if ginkgo fails